Code/Tools/SerializeContextTools/Dumper.cpp

```diff
@@ -596,6 +596,94 @@ namespace AZ::SerializeContextTools
         return result;
     }
 
+    bool Dumper::CreateUuid(Application& application)
+    {
+        // outputStream defaults to writing to stdout
+        AZStd::variant<FunctorStream, AZ::IO::SystemFileStream> outputStream(AZStd::in_place_type<FunctorStream>,
+            GetWriteBypassStdoutCapturerFunctor(application));
+
+        AZ::CommandLine& commandLine = *application.GetAzCommandLine();
+        // If the output-file parameter has been supplied open the file path using FileIOStream
+        if (size_t optionCount = commandLine.GetNumSwitchValues("output-file"); optionCount > 0)
+        {
+            AZ::IO::FixedMaxPath outputPath;
+            if (AZ::IO::PathView outputPathView(commandLine.GetSwitchValue("output-file", optionCount - 1));
+                outputPathView.IsRelative())
+            {
+                AZ::Utils::ConvertToAbsolutePath(outputPath, outputPathView.Native());
+            }
+            else
+            {
+                outputPath = outputPathView.LexicallyNormal();
+            }
+
+            constexpr AZ::IO::OpenMode openMode = AZ::IO::OpenMode::ModeWrite | AZ::IO::OpenMode::ModeCreatePath;
+
+            if (auto& fileStream = outputStream.emplace<AZ::IO::SystemFileStream>(outputPath.c_str(), openMode);
+                !fileStream.IsOpen())
+            {
+                AZ_Printf(
```

Comment:
Nit: Should we make this an error or warning instead?

Reply:
Actually not in the SerializeContext tools.
When a non-zero return code is received from a command it redirect [output to stderr](https://github.com/o3de/o3de/blob/development/Code/Tools/SerializeContextTools/Runner.cpp#L185-L199), which is normal for a CLI application